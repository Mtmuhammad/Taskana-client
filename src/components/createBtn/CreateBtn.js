// Btn to create task, projects, and tickets
import React from 'react'

const CreateBtn = ({ name }) => {
  return (
    <div
      data-testid="btn-container"
      className="mb-5 d-flex justify-content-end"
    >
      <button
        type="button"
        className="btn create-btn"
        data-bs-toggle="modal"
        data-bs-target={`#create${name}`}
      >
        <i data-testid="btn-icon" className="me-2 bx bxs-plus-circle"></i>Create{" "}
        {name}
      </button>
    </div>
  );
};

export default CreateBtn;
