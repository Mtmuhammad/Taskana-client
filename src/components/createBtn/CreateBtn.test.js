import React from "react";
import { render, screen } from "@testing-library/react";
import CreateBtn from "./CreateBtn";

describe("smoke and snapshot tests", () => {
  test("should render without crashing", () => {
    render(<CreateBtn name="Project" />);
  });

  test("should match snapshot", () => {
    const { asFragment } = render(<CreateBtn name="Project" />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("create btn", () => {
  test("should render btn container", () => {
    render(<CreateBtn name="Project" />);
    const container = screen.getByTestId("btn-container");
    expect(container).toBeInTheDocument();
    expect(container).toHaveClass("mb-5 d-flex justify-content-end");
  });
  test("should render btn", () => {
    render(<CreateBtn name="Project" />);
    const btn = screen.getByRole("button");
    expect(btn).toBeInTheDocument();
    expect(btn).toHaveClass("btn create-btn");
  });
  test("should render btn icon", () => {
    render(<CreateBtn name="Project" />);
    const icon = screen.getByTestId("btn-icon");
    expect(icon).toBeInTheDocument();
    expect(icon).toHaveClass("me-2 bx bxs-plus-circle");
  });
});
