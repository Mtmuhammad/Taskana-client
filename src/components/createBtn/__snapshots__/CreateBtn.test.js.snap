// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke and snapshot tests should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="mb-5 d-flex justify-content-end"
    data-testid="btn-container"
  >
    <button
      class="btn create-btn"
      data-bs-target="#createProject"
      data-bs-toggle="modal"
      type="button"
    >
      <i
        class="me-2 bx bxs-plus-circle"
        data-testid="btn-icon"
      />
      Create Project
    </button>
  </div>
</DocumentFragment>
`;
